### PR TITLE
Feature/rule updates

### PR DESCRIPTION
**Pull request by rosshadden**
`/needs` #1 

Updated linting rules.

### CODE REVIEW

File: frameworks/ember.js [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
+		'ember-cleanup/no-this-in-dep-keys': 2,
+		'ember-cleanup/no-throw': 0, // We prefer throwing XErrors
+		'ember-cleanup/no-typeof': [ 2, { 'disallowed': [ 'object' ] } ],
+		'ember-cleanup/no-typo-in-dep-keys': [ 1, { 'ignoreExclamationMark': true } ],
+		'ember-cleanup/one-level-each': 2,
+		'ember/ember-extend-comments': [ 2, [ 'actions', 'classNames' ] ],
+		'ember/ember-newline-extend': 2,
+		'ember/ember-newline-reopenclass': 2,
+		'ember/ember-reopenclass-comments': [ 2, [] ],
+		'indent': [ 2, 2, { 'SwitchCase': 1 } ]
```
Comment (rosshadden) [inline frameworks/ember.js]: The only actual change made to the ember rules (other than sorting the keys like they now are in every other file, which will probably cause noise) was moving `indent` from `browser.js` to `ember.js`.  The motivation for this is pretty clear.

**Comment by rosshadden** [comment]
`/cc` @crispy1989 @AO16

**Comment by rosshadden** [comment]
Bump!

**Comment by rjhilgefort** [comment]
Needs updates that preserve a lint error free run in pulse/pulse api.

**Pull request merged**